homeassistant/components/nest/camera.py

```diff
@@ -1,19 +1,228 @@
-"""Support for Nest cameras that dispatches between API versions."""
+"""Support for Google Nest SDM Cameras."""
+from __future__ import annotations
 
+import asyncio
+from collections.abc import Callable
+import datetime
+import functools
+import logging
+from pathlib import Path
+
+from google_nest_sdm.camera_traits import (
+    CameraImageTrait,
+    CameraLiveStreamTrait,
+    RtspStream,
+    StreamingProtocol,
+)
+from google_nest_sdm.device import Device
+from google_nest_sdm.device_manager import DeviceManager
+from google_nest_sdm.exceptions import ApiException
+
+from homeassistant.components.camera import Camera, CameraEntityFeature, StreamType
+from homeassistant.components.stream import CONF_EXTRA_PART_WAIT_TIME
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity import DeviceInfo
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.event import async_track_point_in_utc_time
+from homeassistant.util.dt import utcnow
+
+from .const import DATA_DEVICE_MANAGER, DOMAIN
+from .device_info import NestDeviceInfo
+
+_LOGGER = logging.getLogger(__name__)
 
-from .camera_sdm import async_setup_sdm_entry
-from .const import DATA_SDM
-from .legacy.camera import async_setup_legacy_entry
+PLACEHOLDER = Path(__file__).parent / "placeholder.png"
+
+# Used to schedule an alarm to refresh the stream before expiration
+STREAM_EXPIRATION_BUFFER = datetime.timedelta(seconds=30)
 
 
 async def async_setup_entry(
```

Comment:
Reviewer note: This file is a rename of `camera_sdm.py` with this method renamed.